lib/src/components/NxButton/NxButton.tsx

```diff
@@ -14,7 +14,7 @@ const NxButton = forwardRef<HTMLButtonElement, Props>(
       const classNames = classnames('nx-btn', className, {
         // secondary is the default, its styles are directly on `nx-btn`
         [`nx-btn--${variant}`]: variant && variant !== 'secondary',
-        'nx-btn--icon-only': iconOnly,
+        'nx-btn--tertiary nx-btn--icon-only': iconOnly,
```

Comment:
If something like this is constructed: `<NxButton iconOnly variant="primary"/>`, both the `.nx-btn--primary` and `.nx-btn-tertiary` classes would end up on the `<button>`.  Whereas previously we wanted to allow icon-only buttons of any variant, and thus had `iconOnly` as a separate prop, I think now what we want is for `iconOnly` to be one of the `variant` values and to get rid of the separate prop.  This would prevent the caller from using `iconOnly` and a different, incompatible `variant` at the same time.

Reply:
> If something like this is constructed: `<NxButton iconOnly variant="primary"/>`, both the `.nx-btn--primary` and `.nx-btn-tertiary` classes would end up on the `<button>`. Whereas previously we wanted to allow icon-only buttons of any variant, and thus had `iconOnly` as a separate prop, I think now what we want is for `iconOnly` to be one of the `variant` values and to get rid of the separate prop. This would prevent the caller from using `iconOnly` and a different, incompatible `variant` at the same time.

I knew there was something fishy there but needed your comment to fully grok it: https://github.com/sonatype/sonatype-react-shared-components/pull/138/commits/b1733248a4de3895fc82da9fdad044f2cc2dcd4c